pyiceberg/table/__init__.py

```diff
@@ -1064,6 +1067,81 @@ def name_mapping(self) -> Optional[NameMapping]:
         """Return the table's field-id NameMapping."""
         return self.metadata.name_mapping()
 
+    @dataclass
+    class UpsertResult:
+        """docstring"""
+        rows_updated: int
+        rows_inserted: int
+        info_msgs: str
+        error_msgs: str
```

Comment:
I think we can do a bit better here. We can set some defaults and make [it frozen](https://docs.python.org/3/library/dataclasses.html#frozen-instances). 
```suggestion
    @dataclass(frozen=True)
    class UpsertResult:
        """Summary the upsert operation"""
        rows_updated: int = 0
        rows_inserted: int = 0
        info_msg: Optional[str] = None
        error_msg: Optional[str] = None
```

Reply:
done; thanks